Extend Type fails if type implements Node, depending on file alphabetical order

When extending a type that `implements Node` and if the file that extends the type comes before the file that declares the type in alphabetical order, then the merge fails.

Person.graphql
```graphql
type Person implements Node {
  name: String!
}
```

ExtendedPerson.graphql
```graphql
extend type Person {
  age: Int!
}
```

For the schemas above, since `ExtendedPerson` comes before `Person` in alphabetical order, the merge will fail. If we rename `ExtendedPerson` to `zExtendedPerson` it will come after `Person` in alphabetical order and the merge will succeed.

It's important to note that the order of the files only matters when the type `implements Node`, otherwise the merge succeeds as expected.

Repro command:

```
go run github.com/mununki/gqlmerge --indent=2s test/*.graphql test/t/s.graphql
```

Error:
```
🎉 [test/extendedPerson.graphql] Total 1 GraphQL files found!
🎉 [test/person.graphql] Total 1 GraphQL files found!
panic: lib.Error("Duplicated Types: Person(test/extendedPerson.graphql:1:12) and (test/person.graphql:1:5)")

goroutine 36 [running]:
github.com/mununki/gqlmerge/lib.errorf(...)
        /go/pkg/mod/github.com/mununki/gqlmerge@v0.2.13/lib/parse.go:14
github.com/mununki/gqlmerge/lib.(*Schema).MergeTypeName(0x1400012e1c0, 0x0?)
        /go/pkg/mod/github.com/mununki/gqlmerge@v0.2.13/lib/schema.go:565 +0x6ec
created by github.com/mununki/gqlmerge/lib.mergeSchemas in goroutine 1
        /go/pkg/mod/github.com/mununki/gqlmerge@v0.2.13/lib/merge.go:77 +0x88c
```

--- a/lib/schema.go
+++ b/lib/schema.go
@@ -5,6 +5,7 @@ import (
 	"os"
 	"path/filepath"
 	"reflect"
+	"sort"
 	"sync"
 )
 
@@ -538,6 +539,9 @@ func (s *Schema) MergeTypeName(wg *sync.WaitGroup) {
 	defer wg.Done()
 	j := 0
 	seen := make(map[string]struct{}, len(s.Types))
+	sort.SliceStable(s.Types, func(i, j int) bool {
+		return !s.Types[i].Extend && s.Types[j].Extend
+	})
 	for _, v := range s.Types {
 		if _, ok := seen[v.Name]; ok {
 			for i := 0; i < j; i++ {

--- a/test/object_extension/schema/a.graphql
+++ b/test/object_extension/schema/a.graphql
@@ -1,6 +1,3 @@
-type Person implements Node @talkable{
-  id: ID!
-  createTime: Time!
-  updateTime: Time!
-  name: String!
+extend type Person @walkable {
+  hasCar: Boolean!
 }

--- a/test/object_extension/schema/b.graphql
+++ b/test/object_extension/schema/b.graphql
@@ -1,3 +1,6 @@
-extend type Person @walkable {
-  hasCar: Boolean!
+type Person implements Node @talkable{
+  id: ID!
+  createTime: Time!
+  updateTime: Time!
+  name: String!
 }

